config/config.go

```diff
@@ -776,3 +782,39 @@ func (re Regexp) MarshalJSON() ([]byte, error) {
 	}
 	return nil, nil
 }
+
+// Matchers is label.Matchers with an added UnmarshalYAML method to implement the yaml.Unmarshaler interface.
+type Matchers labels.Matchers
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for Matchers.
+func (m *Matchers) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var lines []string
+	if err := unmarshal(&lines); err != nil {
+		return err
+	}
+	for _, line := range lines {
+		pm, err := labels.ParseMatchers(line)
+		if err != nil {
+			return err
+		}
+		*m = append(*m, pm...)
+	}
+	sort.Sort(labels.Matchers(*m))
+	return nil
+}
+
+// MarshalYAML implements the yaml.Marshaler interface for Matchers.
+func (m *Matchers) MarshalYAML() (interface{}, error) {
+	var buf bytes.Buffer
+
+	buf.WriteByte('{')
+	for i, ms := range *m {
+		if i > 0 {
+			buf.WriteByte(',')
+		}
+		buf.WriteString(ms.String())
+	}
+	buf.WriteByte('}')
+
+	return buf.String(), nil
```

Comment:
This is just replicating the code in `Matchers.String`. You can just write
```Go
    return m.String(), nil
```
instead.

However, since we are marshaling a Yaml list type here, we have to return a slice to make it work properly. You could return `[]string{m.String()}`, but I would just put one matcher into each element of the Yaml list, i.e. do the following:
```Go
func (m Matchers) MarshalYAML() (interface{}, error) {
	result := make([]string, len(m))
	for i, matcher := range m {
		result[i] = matcher.String()
	}
	return result, nil
}
```